cardano-cli/src/Cardano/CLI/Shelley/Util.hs

```diff
@@ -0,0 +1,32 @@
+-- | Shelley CLI shared Utility functions
+module Cardano.CLI.Shelley.Util
+  (
+    displayErrorText
+  , rescue
+
+  , getNetworkId
+  , getNetworkId'
+  ) where
+
+import           Control.Monad.Trans.Except.Extra (firstExceptT, newExceptT)
+import           Data.Text as Text (pack)
+
+import           Cardano.Api
+import           Cardano.Prelude
+
+import           Cardano.CLI.Shelley.Commands (NetworkIdArg(..) )
+
+displayErrorText :: Error e => e -> Text
+displayErrorText = Text.pack . displayError
+
+rescue :: Functor m =>  m (Either x a) -> (x -> y) -> ExceptT y m a
```

Comment:
It is called `rescue` in Ruby.
I like `rescue` much better than `firstNewExceptT` . 
Also it is more readable to add  " \`rescue\` ErrorTag" trailing at the end of the line.
Then the good- case comes first and error handling is less distracting when reading the code.
(The `rescue` pattern exists in several other languages..)

Reply:
`rescue` is still vague IMO and because it exists in other languages doesn't necessarily mean it's a good name. What about `liftMapError`? Let's get @newhoggy 's input on this

Reply:
For me `rescue` is much easier to read and type than `liftMapError`. 7 vs 12 char and no need for Caps.
Some Haskell guys would even introduce a `&%>` style operator for that.
Best would be the input of @newhoggy in 30 Years, or some old coder, who needs thick glasses and uses ever bigger font sizes.
Readability is important.
Some IDEs even show the type when hovering the function.

Reply:
This comment is a bit tangential but relevant.

I had something thoughts about error handling in general.

A function like `rescue` seems generic - but then if it is generic it should go in some generic error handling module, but when we do that we might eventually encounter a problem:

A function like `rescue` seems fine by itself, but in practise, we might need a family of functions and would then have trouble naming them.  For example:

```
rescue :: Functor m =>  m (Either x a) -> (x -> y) -> ExceptT y m a

rescue :: Functor m =>  Either x a -> (x -> y) -> ExceptT y m a

rescue :: Functor m =>  ExceptT x m a -> (x -> y) -> ExceptT y m a
```

I was unhappy with my earlier PR that introduced `onNothingThrow` for this reason.  It was defined like this:

```
onNothingThrow :: Monad m => e -> Maybe a -> ExceptT e m a
```

But then what about all these other possibilities?

```
onNothingThrow :: Monad m => e -> Maybe a -> ExceptT e m a

onNothingThrow :: Monad m => e -> m (Maybe a) -> ExceptT e m a

onNothingThrow :: Monad m => e -> ExceptT e m (Maybe a) -> ExceptT e m a
```

I decided I only need the last one because everything else can be lifted to it.

See this PR: https://github.com/input-output-hk/cardano-node/pull/4807

Interested in your thoughts.

(Note, I'm intending to restructure `oops` to use more readable English names and avoid the explosion of permutations in a similar way)

Reply:
Nevermind I close the PR and favouring `hoistMaybe` instead.

Reply:
I suggest we don't define `rescue` and should instead inline it:

```
getNetworkId :: NetworkIdArg -> (EnvNetworkIdError -> e) -> ExceptT e IO NetworkId
getNetworkId arg errorTag
  = firstExceptT errorTag $ ExceptT $ getNetworkId' arg
```

Reply:
I agree with defining it inline.

Reply:
There is not one use site of the `rescue` function but there are about ca 100 possible use sites.
This is the 101 of programming: if there are multiple repetitions of the same pattern one defines a function.
Otherwise the code becomes extremely ugly and hard to maintain.
(As is already the case.)
If there is a need for a family of functions which all do something similar like lifting and error handling
then that is indeed a case for a type class.
It makes no sense at all to just refuse to add any kind of abstraction, just because people cannot agree on a common name for a function.
Once an abstraction is in place, it is very easy to rename the function.
It is better to start with a simple abstraction and think about a type classs later.

Reply:
> I had something thoughts about error handling in general.
> A function like rescue seems generic - but then if it is generic it should go in some generic error handling
> module, but when we do that we might eventually encounter a problem:
> A function like rescue seems fine by itself, but in practise, we might need a family of functions and would then
> have trouble naming them. For example:

This kind of argument is well know and it goes by the phrase:
"Perfect is the enemy of good"

As stated earlier, there are lots and lots of places in the code that could be simplified by using `rescue`.
If there is need for a more elegant and more generic solution in the future
then replacing the repetitive pattern with a function like `rescue` now is still a very good step in that direction.